src/org/openstreetmap/josm/gui/layer/imagery/ColorfulFilter.java

```diff
@@ -212,11 +212,7 @@ private int mixInt(int color, double luminosity) {
         int val = (int) (colorfulness * color + (1 - colorfulness) * luminosity);
         if (val < 0) {
             return 0;
-        } else if (val > 0xff) {
-            return 0xff;
-        } else {
-            return val;
-        }
+        } else return Math.min(val, 0xff);
```

Comment:
I don't know about this. It almost seems like it should be `Math.max(0, Math.min(0xff, val))`.

Maybe add a comment indicating that we should use `Math.clamp` when we get to Java 21. :)